src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.ref.cs

```diff
@@ -60,7 +60,7 @@ public static partial class CacheEntryExtensions
     public static partial class CacheExtensions
```

Comment:
Do these files need to go into the `src` directory? That breaks the DefaultCompileItems stuff that we need to work around. And it is kind of odd to have `.cs` files in the `src` directory that aren't compiled into the project.

Reply:
What would be a better location for these files? Also should we make it clearer that these files are auto-generated and shouldn't be mutated? I imagine we could update the file header to call that out and also name the files differently, i.e. $(MsBuildProjectName).ref.g.cs or $(MsBuildProjectName).ref.generated.cs.

Reply:
What about leaving them in the `ref` folder?

Agreed - updating the header and naming the files to indicate they are generated would be useful.

Reply:
> That breaks the DefaultCompileItems stuff that we need to work around.

Actually this doesn't break anything, does it? In the first commit I explicitly added that file to the exclusion list of glob patterns.

> What about leaving them in the ref folder?

IMO having a dedicated "ref" folder just for this single file is weird... But I don't have a better suggestion aside from what I'm proposing here (src folder).

Reply:
I don't like having `.cs` files inside the `src` folder that aren't compiled into the project. I think it is confusing.

You mention in the OP that

> It feels right to produce reference assemblies via Roslyn for all tfms

If we ever wanted to do this in the future, we would have more than a single file. I imagine we could have one ref `.cs` file for each TFM that we target. (And maybe if the APIs aren't different between TFMs, then just a single file.)

Either way, these files aren't part of the `src` project, so I wouldn't put them in the `src` folder.

Reply:
I agree with @eerhardt.  These aren't src.  I don't see why keeping them in ref is a problem.